fs/fs_zenfs.cc

```diff
@@ -800,6 +827,38 @@ IOStatus ZenFS::RenameFile(const std::string& source_path,
   return s;
 }
 
+IOStatus ZenFS::LinkFile(const std::string& src, const std::string& dst,
+                         const IOOptions& options, IODebugContext* dbg) {
+  std::shared_ptr<ZoneFile> src_file(nullptr);
+  std::shared_ptr<ZoneFile> dst_file(nullptr);
+  IOStatus s;
+
+  Debug(logger_, "LinkFile: %s to %s\n", src.c_str(), dst.c_str());
+
+  /* Return if specified link already exists for the src file */
+  dst_file = GetFile(dst);
+  if (dst_file != nullptr)
+    return IOStatus::InvalidArgument("Failed to create link, link exists");
+
+  src_file = GetFile(src);
+  if (src_file != nullptr) {
+    src_file->SetLinkInfo(dst.c_str());
```

Comment:
Probalby does not break something:
Can we move `src_file->SetLinkInfo(dst.c_str());` also in the section where `file_mtx_` is locked? `DeleteFile` also has the `file_mtx_` before it modifies links.

Reply:
SetLinkInfo() does get the zonefile lock before updating, so should be ok here. The file_mtx_ lock is for the file system, so i did not think it was needed here.  I will move after the lock is held.